lib/facter/util/resolvers/ffi/hostname.rb

```diff
@@ -36,21 +36,32 @@ def self.getffihostname
             raw_hostname.read_string
           end
 
-          def self.getffiaddrinfo(hostname)
-            hostname_ptr = FFI::MemoryPointer.new(hostname)
+          def self.getffiaddrinfo(hostname) # rubocop:disable  Metrics/AbcSize
+            ret = FFI::MemoryPointer.new(:pointer)
```

Comment:
could we make sure that the new pointer is NULL (or check, it may be default to NULL from ruby)